### DISCUSSION TITLE
Environment setup

### DISCUSSION THREAD

**Discussion by zmd4** [GunnarMorrigan/mqrstt#17] [Q&A] [answered]
I'm new to both Rust and MQTT, and I'm having a hard time compiling the example code.  Are there any environment "gotchyas"?  I'm working with Rust 1.75 on Rocky Linux.

On the Tokio example, I get the `lifetimes do not match method in trait` error on the `handle()` definition.

**Comment by GunnarMorrigan** [comment]
Hi,

I am not sure if the latest examples are up to date on master.

The master branch is under active development (alpha versions) to also allow parallel reads and writes. To increase speed on backend services when needed.

What runtime are you using? 
Serial or concurrent driver?
TLS or TCP? 
There were also some changes in TLS libraries.

I will update the examples to match the current implementation.

Cheers :)

**Reply by zmd4** [reply]
Ah, that makes sense.  I think I'm most interested in the tokio TCP example at this point.

These are my runtimes:
```
[dependencies]
tokio = { version = "1.35.1", features = ["rt-multi-thread", "rt", "macros", "sync", "io-util", "net", "time"] }
#tokio-rustls = "0.25.0"
async-trait = "0.1.77"
bytes = "1.5.0"
mqrstt = { version = "0.2.2", default-features = false, features = ["tokio"]}
```
Thanks!

**Reply by GunnarMorrigan** [reply, marked as answer]
Could you post your code in a repo example?

I will actively back this crate. It is used in production at my company. 

I will also update the examples. Are you sure that you are not trying examples on master branch which is mqrstt version 0.3.0-alpha while you run with 'mqrstt' version 0.2.2.

**Reply by zmd4** [reply]
Yeah, I'm pretty sure that I am the problem. lol  When looking at MQTT and Rust this repo seems to come up a lot.

Oh yeah, I am copying them from the master branch here.  What is the commit hash for 0.2.2?

My repo: https://github.com/zmd4/router_health

**Reply by GunnarMorrigan** [reply]
Hi,

I tagged and created a release for v0.2.2:
https://github.com/GunnarMorrigan/mqrstt/releases/tag/v0.2.2

Also I added v0.2.2 examples for both TCP and TLS to the main branch here:
https://github.com/GunnarMorrigan/mqrstt/tree/main/examples


Note the TLS examples use deprecated functions. I have already fixed this but need to add it upstream.

**Reply by zmd4** [reply]
Yes, this was the problem I made.  Using the 0.2.2 examples it compiles.  Thank you for your help and patience Gunnar!
